### ISSUE TITLE
Request for your help regarding dependency issues

### ISSUE THREAD

**Issue by mahmoudzamani** [decile-team/submodlib#31]
Error:

Building wheels for collected packages: submodlib
  Building wheel for submodlib (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\\Anaconda3\envs\DistilP38\python.exe' 'C:\Users\\Anaconda3\envs\DistilP38\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' build_wheel 'C:\Us\AppData\Local\Temp\tmp0bxa4jgu'
       cwd: C:\Users\AppData\Local\Temp\pip-install-_xv0spoi\submodlib_2bb9afc947884ca9b86f9154ca4eb84d
  Complete output (43 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-cpython-38
  creating build\lib.win-amd64-cpython-38\submodlib
  copying submodlib\helper.py -> build\lib.win-amd64-cpython-38\submodlib
  copying submodlib\myfunctions.py -> build\lib.win-amd64-cpython-38\submodlib
  copying submodlib\version.py -> build\lib.win-amd64-cpython-38\submodlib
  copying submodlib\__init__.py -> build\lib.win-amd64-cpython-38\submodlib
  creating build\lib.win-amd64-cpython-38\submodlib\functions
  copying submodlib/functions\clustered.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\concaveOverModular.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\disparityMin.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\disparitySum.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\facilityLocation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\facilityLocationConditionalGain.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\facilityLocationConditionalMutualInformation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\facilityLocationMutualInformation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\facilityLocationVariantMutualInformation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\featureBased.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\graphCut.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\graphCutConditionalGain.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\graphCutMutualInformation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\logDeterminant.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\logDeterminantConditionalGain.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\logDeterminantConditionalMutualInformation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\logDeterminantMutualInformation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\mixture.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\probabilisticSetCover.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\probabilisticSetCoverConditionalGain.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\probabilisticSetCoverConditionalMutualInformation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\probabilisticSetCoverMutualInformation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\saturatedCoverage.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\setCover.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\setCoverConditionalGain.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\setCoverConditionalMutualInformation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\setCoverMutualInformation.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\setFunction.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  copying submodlib/functions\__init__.py -> build\lib.win-amd64-cpython-38\submodlib/functions
  running build_ext
  building 'submodlib_cpp' extension
  error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
  ----------------------------------------
  ERROR: Failed building wheel for submodlib
Failed to build submodlib
ERROR: Could not build wheels for submodlib which use PEP 517 and cannot be installed directly

================

my pip list:

Package                       Version
----------------------------- -----------
alabaster                     0.7.12
attrs                         19.3.0
autograd                      1.3
autograd-gamma                0.5.0
Babel                         2.10.1
certifi                       2022.5.18.1
charset-normalizer            2.0.12
colorama                      0.4.4
dm-tree                       0.1.6
docutils                      0.18.1
formulaic                     0.2.4
gast                          0.4.0
grpcio                        1.34.1
idna                          3.3
imagesize                     1.3.0
importlib-metadata            4.11.4
interface-meta                1.2.4
Jinja2                        3.1.2
joblib                        1.1.0
latexcodec                    2.0.1
lifelines                     0.26.4
llvmlite                      0.38.1
MarkupSafe                    2.1.1
mkl-fft                       1.3.1
mkl-random                    1.2.2
mkl-service                   2.4.0
multipledispatch              0.6.0
numba                         0.55.2
numpy                         1.20.1
packaging                     21.3
pip                           21.2.2
portpicker                    1.3.9
pybtex                        0.24.0
pybtex-docutils               1.0.2
Pygments                      2.12.0
pyparsing                     3.0.9
pytz                          2022.1
PyYAML                        6.0
requests                      2.28.0
retrying                      1.3.3
scikit-learn                  0.23.0
scipy                         1.4.1
semantic-version              2.8.5
setuptools                    61.2.0
six                           1.16.0
snowballstemmer               2.2.0
Sphinx                        5.0.1
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-bibtex          2.4.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
stg                           0.1.2
tensorflow-addons             0.11.2
tensorflow-estimator          2.5.0
tensorflow-federated          0.17.0
tensorflow-model-optimization 0.4.1
tensorflow-privacy            0.5.2
threadpoolctl                 3.1.0
tqdm                          4.64.0
typeguard                     2.13.3
urllib3                       1.26.9
wheel                         0.37.1
wincertstore                  0.2
zipp                          3.8.0

Thank you

**Comment by surajkothawade** [comment]
Hi Mahmoud,

Building submodlib requires installation of C++14 or greater.

I recommend trying GCC 5.2 or greater.

Thanks!
